src/Stripe.net/Infrastructure/Middleware/ParserPlugin/DictionaryPlugin.cs

```diff
@@ -9,13 +10,25 @@ internal class DictionaryPlugin : IParserPlugin
     {
         public bool Parse(ref string requestString, JsonPropertyAttribute attribute, PropertyInfo property, object propertyValue, object propertyParent)
         {
-            if (!attribute.PropertyName.Contains("metadata") && !attribute.PropertyName.Contains("fraud_details") && !attribute.PropertyName.Contains("attributes")) return false;
+            // Check if the property is a Dictionary
+            var type = property.PropertyType;
+            if (!type.GetTypeInfo().IsGenericType) return false;
+            if (type.GetTypeInfo().GetGenericTypeDefinition() != typeof(Dictionary<,>)) return false;
+
+            // Ensure that key and value types are both string
```

Comment:
There's probably a way to be able to encode any `Dictionary<object, object>` by calling `ToString()` on keys and values, but I couldn't manage with my meager .NET reflection skills, so I'm just throwing an exception if the key and/or value type is not `string`.

It's not a big deal as all dictionaries in the library are `<string, string>` right now and the explicit exception should prevent contributors from mistakenly using other types.

Reply:
Honestly, I think some explicitness is better here than more reflection magic :)

+1 on the `ArgumentException`s.